test/UnionIterator-test.js

```diff
@@ -226,22 +239,9 @@ describe('UnionIterator', () => {
     });
 
     describe('before reading', () => {
-      it('should have read the sources', () => {
-        sourceIterator.read.should.have.been.called;
-      });
-
       it('should not have ended', () => {
         iterator.ended.should.be.false;
       });
-
-      it('should pass errors', () => {
-        const callback = sinon.spy();
-        const error = new Error('error');
-        iterator.once('error', callback);
-        sourceIterator.emit('error', error);
-        callback.should.have.been.calledOnce;
-        callback.should.have.been.calledWith(error);
-      });
```

Comment:
This is an example of an error that can no longer be propogated with the changes we have made.